src/components/media-views.js

```diff
@@ -750,31 +752,40 @@ AFRAME.registerComponent("media-image", {
         }
       }
 
+      let cacheItem;
       if (textureCache.has(src)) {
-        const cacheItem = textureCache.retain(src);
-        texture = cacheItem.texture;
-        ratio = cacheItem.ratio;
+        cacheItem = textureCache.retain(src);
       } else {
         if (src === "error") {
-          texture = errorTexture;
-        } else if (contentType.includes("image/gif")) {
-          texture = await createGIFTexture(src);
-        } else if (contentType.startsWith("image/")) {
-          texture = await createImageTexture(src);
+          cacheItem = errorCacheItem;
+        } else if (inflightTextures.has(src)) {
+          await inflightTextures.get(src);
+          cacheItem = textureCache.retain(src);
         } else {
-          throw new Error(`Unknown image content type: ${contentType}`);
+          let promise;
+          if (contentType.includes("image/gif")) {
+            promise = createGIFTexture(src);
+          } else if (contentType.startsWith("image/")) {
+            promise = createImageTexture(src);
+          } else {
+            throw new Error(`Unknown image content type: ${contentType}`);
+          }
+          inflightTextures.set(src, promise);
+          texture = await promise;
```

Comment:
This isn't a concern that is new with your changes, but if a `media-image`'s `entity` gets removed after `createImageTexture` is called, but before the `promise` resolves, then will `_hasRetainedTexture` be `false` when checked in `remove`? (If so, does than mean we'll retain and never release this texture?)

Reply:
Good catch, added a fix